src/PaymentChannel.js

```diff
@@ -0,0 +1,114 @@
+import BigNumber from 'bignumber.js';
+
+import paymentChannelStateMessages from './payment_channel_state_service_pb';
+
+export default class PaymentChannel {
+  constructor(channelId, web3, account, paymentChannelStateServiceClient, mpeContract) {
+    this._channelId = channelId;
+    this._web3 = web3;
+    this._account = account;
+    this._mpeContract = mpeContract;
+    this._paymentChannelStateServiceClient = paymentChannelStateServiceClient;
+    this._state = {
+      nonce: new BigNumber(0),
+      lastSignedAmount: new BigNumber(0),
+    };
+  }
+
+  get channelId() {
+    return this._channelId;
+  }
+
+  get nonce() {
+    return this.state.nonce;
+  }
+
+  get lastSignedAmount() {
+    return this.state.lastSignedAmount;
+  }
+
+  get state() {
+    return this._state;
+  }
+
+  updateState(gstate) {
+    this._state = {
+      ...this._state,
+      ...state,
+    }
+  }
+
+  hasSufficientFunds(amount) {
+    return this.state.availableAmount >= amount;
+  }
+
+  isValid(expiry) {
```

Comment:
This specific method could be moved into `PaymentChannelManagementStrategy`.

Reply:
Wouldn't it be handy to have them on the `PaymentChannel` so that we could make decision by asking the payment channel whether it is expired or valid? What we would have to do in `PaymentChannelManagementStrategy` otherwise is something like this 
`paymentChannel.state.expiration > expiry` for expiry or `paymentChannel.state.availableAmount > amount` for funds. We could move it out of `PaymentChannel` if we thing the meaning of `hasSufficientFunds` and `isValid` would change depending on how the developer wants to treat it.

Reply:
My concern here is that `hasSufficientFunds` answers to only question `paymentChannel.state.availableAmount > amount`. If developer has another condition to check it anyway should use `paymentChannel.state.availableAmount` directly. So `hasSufficientFunds` is special method which could be easily implemented outside and handles only one case. The same is for `isValid`.

If your concern is that time is really complex matter to work with, I agree. But I think it could be solved using some object instead of plain number to represent the time. Like it is implemented in `Java` or `Golang` time libraries.

Reply:
After rethinking a bit, I agree with you on this and have removed these methods.

Reply:
Addressed in [commit](https://github.com/singnet/snet-sdk-js/pull/1/commits/1adf2e3c86a29e3437e5216bb44a22f035277728)